8.15.0 Release notes (#5602)
elastic/security-docs: docs/release-notes/8.15.asciidoc

[[release-notes-header-8.15.0]]
== 8.15.0

[discrete]
[[release-notes-8.15.0]]
=== 8.15.0

[discrete]
[[breaking-changes-8.15.0]]
==== Breaking changes

* User-defined quick prompts for Elastic AI Assistant that were created before you upgrade to 8.15 will no longer appear after you upgrade. In 8.15, quick prompts will be shared by all users in your deployment, rather than stored at the user level. To implement your existing quick prompts in 8.15, copy them prior to upgrade, and add them again after upgrade ({kibana-pull}187040[#187040]).

[discrete]
[[features-8.15.0]]
==== New features

* Adds the `scan` action to the response console, which allows you to scan a specific file or directory on a host for malware ({kibana-pull}184723[#184723]).

> * Adds a Defend Advanced Policy option that allows you to opt out of registry event filtering ({kibana-pull}186564[#186564]).

Reviewer: ```suggestion
* Adds an {elastic-defend} integration policy option in Advanced Settings that allows you to opt out of registry event filtering ({kibana-pull}186564[#186564]).
```

* Allows you to specify additional file and registry paths to monitor for read access ({kibana-pull}181361[#181361]).

> * Allows you to use {elastic-sec} to isolate and release hosts running a Crowdstrike agent ({kibana-pull}186801[#186801]).

Reviewer: ```suggestion
* Allows you to use {elastic-sec} to isolate and release hosts running a CrowdStrike agent ({kibana-pull}186801[#186801]).
```

> * Allows you to use {elastic-sec} to isolate and release hosts running a CrowdStrike agent ({kibana-pull}186801[#186801]).

Author: No objections to this placement. Thanks for adding these in!

> * Updates entity risk scores when asset criticality changes ({kibana-pull}182234[#182234]).

Reviewer 2: ```suggestion
* Recalculates entity risk scores when asset criticality changes on an individual entity ({kibana-pull}182234[#182234]).
```
Since this doesn't apply to bulk assignment yet.

* Provides previews of hosts, users, and alerts that you're examining in the alert details flyout ({kibana-pull}186850[#186850], {kibana-pull}186857[#186857]).
* Adds an option for toggling row renderers on and off, and moves notes to a new flyout in Timeline ({kibana-pull}186948[#186948]).

[discrete]
[[enhancements-8.15.0]]
==== Enhancements

* Adds notifications to the connector dropdown menu on the **Attack discovery** page so you know when other connectors have new discoveries ({kibana-pull}186903[#186903], {kibana-pull}187209[#187209]).
* Improves the output of discoveries generated with `GPT-4o` ({kibana-pull}187183[#187183]).
* Allows Attack discovery generation to continue when you navigate to another page ({kibana-pull}184949[#184949]).
* Improves AI Assistant responses across multiple connectors and in multiple scenarios, for both streaming and non-streaming use cases ({kibana-pull}182041[#182041]).
* Updates the default Gemini version to `gemini-1.5-pro-001` and the default Bedrock version to `anthropic.claude-3-5-sonnet-20240620-v1:0` ({kibana-pull}186671[#186671]).

> * Introduces a new Endpoint policy setting that allows you to control whether the kernel reports Windows network events that happened on a local loopback interface ({kibana-pull}181753[#181753]).

Reviewer: ```suggestion
* Introduces a new {elastic-endpoint} policy setting that allows you to control whether the kernel reports Windows network events that happened on a local loopback interface ({kibana-pull}181753[#181753]).
```

[[release-notes-header-8.15.0]]
== 8.15.0

[discrete]
[[release-notes-8.15.0]]

> === 8.15.0

Author: ```suggestion
=== 8.15
```

* Allows you to edit the `max_signals`, `related_integrations`, and `required_fields` fields for custom rules ({kibana-pull}179680[#179680], {kibana-pull}178295[#178295], {kibana-pull}180682[#180682]).
* Provides help from AI Assistant when you're correcting rule query errors ({kibana-pull}179091[#179091]).  
* Allows you to bulk update custom highlighted fields for rules ({kibana-pull}179312[#179312]).
* Adds alert suppression for {ml} and {esql} rules ({kibana-pull}181926[#181926], {kibana-pull}180927[#180927]).
* Provides previews of hosts, users, and alerts that you're examining in the alert details flyout ({kibana-pull}186850[#186850], {kibana-pull}186857[#186857]).
* Adds an option for toggling row renderers on and off, and moves notes to a new flyout in Timeline ({kibana-pull}186948[#186948]).

[discrete]
[[enhancements-8.15.0]]
==== Enhancements

* Adds notifications to the connector dropdown menu on the **Attack discovery** page so you know when other connectors have new discoveries ({kibana-pull}186903[#186903], {kibana-pull}187209[#187209]).
* Improves the output of discoveries generated with `GPT-4o` ({kibana-pull}187183[#187183]).

> * Allows Attack discovery generation to continue when you navigate to another page ({kibana-pull}184949[#184949]).

Reviewer 3: You can also run discoveries on multiple connectors simultaneously. The attack discoveries are saved per user and they not only work across another page, but another browser. I would put this note ahead of the one that reads "Adds notifications to the connector dropdown menu on the Attack discovery..."

Author: Thanks for the review, @stephmilovic. Is there a PR tied to the functionality that you're describing? We normally add a link to the PR at the end of the change description so users can check it if they want to learn more.

Reviewer 3: https://github.com/elastic/kibana/pull/184949

* Provides help from AI Assistant when you're correcting rule query errors ({kibana-pull}179091[#179091]).  
* Allows you to bulk update custom highlighted fields for rules ({kibana-pull}179312[#179312]).
* Adds alert suppression for {ml} and {esql} rules ({kibana-pull}181926[#181926], {kibana-pull}180927[#180927]).
* Provides previews of hosts, users, and alerts that you're examining in the alert details flyout ({kibana-pull}186850[#186850], {kibana-pull}186857[#186857]).
* Adds an option for toggling row renderers on and off, and moves notes to a new flyout in Timeline ({kibana-pull}186948[#186948]).
* Creates an LLM connector for Google Gemini ({kibana-pull}183668[#183668]).
* Adds an API for Elastic AI Assistant ({kibana-pull}184485[#184485]).

[discrete]
[[enhancements-8.15.0]]
==== Enhancements

* Allows Attack discovery generation to continue when you navigate to another page, and allows you to run Attack Discovery with multiple connectors at the same time ({kibana-pull}184949[#184949]).

> * Adds notifications to the connector dropdown menu on the **Attack discovery** page so you know when other connectors have new discoveries ({kibana-pull}186903[#186903], {kibana-pull}187209[#187209]).

Reviewer 4: ```suggestion
* Adds notifications to the connector dropdown menu on the Attack discovery page so you know when other connectors have new discoveries ({kibana-pull}186903[#186903], {kibana-pull}187209[#187209]).
```

[[release-notes-header-8.15.0]]
== 8.15

[discrete]
[[release-notes-8.15.0]]
=== 8.15.0

[discrete]
[[known-issue-8.15.0]]
==== Known issues

> // tag::known-issue-[189676][]

Author: Doesn't affect the output. Just cleaning this up.
```suggestion
// tag::known-issue-189676[]
```

[discrete]
.Tags appear in Elastic AI Assistant's responses
[%collapsible]
====
*Details* +
On August 1, 2024, it was discovered that Elastic AI Assistant's responses when using Bedrock Sonnet 3.5 may include `<antThinking>` tags, for example `<search_quality_reflection>` ({kibana-pull}189676[#189676]).

> *Workaround* +

Reviewer 4: ```suggestion
```

> *Workaround* +

Reviewer 4: I agree, it's pretty unnecessary. Was trying a little too hard to follow the standard structure

> Anticipate that tags may appear in AI Assistant's responses.

Reviewer 4: ```suggestion
```

* Updates the default Gemini version to `gemini-1.5-pro-001` and the default Bedrock version to `anthropic.claude-3-5-sonnet-20240620-v1:0` ({kibana-pull}186671[#186671]).
* Simplifies the process of enabling AI Assistant's knowledge base ({kibana-pull}182763[#182763]).
* Unifies the AI Assistant's settings view ({kibana-pull}184678[#184678]).
* Introduces a new {elastic-endpoint} policy setting that allows you to control whether the kernel reports Windows network events that happened on a local loopback interface ({kibana-pull}181753[#181753]).
* Improves how failure messages for the `scan` action appear in the response console ({kibana-pull}186284[#186284]).
* Improves the risk engine's performance. Now, after you turn on the engine, risk data is available sooner ({kibana-pull}184797[#184797]).
* Enhances the risk engine's normalization accuracy ({kibana-pull}184638[#184638]).
* Updates the copy for bulk assigning asset criticality to multiple entities ({kibana-pull}181390[#181390]).
* Improves visual and logic issues in the Findings table ({kibana-pull}184185[#184185]).

> * Enables the expandable alert details flyout by default and replaces the `securitySolution:enableExpandableFlyout` advanced setting with a feature flag that allows you to revert to the old version of the flyout ({kibana-pull}184169[#184169]).

Reviewer 5: ```suggestion
* Enables the expandable alert details flyout by default and replaces the `securitySolution:enableExpandableFlyout` advanced setting with a feature flag that allows you to revert to the old flyout version ({kibana-pull}184169[#184169]).
```

* Fixes the alignment of the page selector dropdown menu on the Shared Exception Lists page ({kibana-pull}187956[#187956]).
* Fixes a rule execution error that occurred when {esql} rules queried source documents with non-ECS compliant sub-fields under the `event.action` field ({kibana-pull}187549[#187549]).
* Fixes a bug that caused the `Enable entity risk scoring` option to display even when you didn't have the correct requirements ({kibana-pull}183517[#183517]).
* Prevents `maxClauseCount` errors from occurring for indicator match rules ({kibana-pull}179748[#179748]).
* Fixes a bug that prevented threat intelligence fields from correctly rendering in the alert details flyout if they had flattened fields ({kibana-pull}179395[#179395]).
* Removes references in the UI that directed users to outdated documentation for the risk scoring feature ({kibana-pull}187585[#187585]).
* Fixes a bug on the Get started page that prevented the correct username from being displayed in the greeting message ({kibana-pull}180670[#180670]).
* Fixes a bug that caused the pagination menu from appearing in the correct place for the Uncommon processes table ({kibana-pull}189201[#189201]).

> * Fixes a bug that affected the panel showing last command details in the Uncommon processes table ({kibana-pull}187848[#187848]).

Reviewer 5: ```suggestion
* Fixes a bug that affected the panel showing the last command details in the Uncommon processes table ({kibana-pull}187848[#187848]).
```
